packages/uploadthing/src/client.ts

```diff
@@ -136,58 +166,50 @@ const uploadFile = <
   },
   presigned: PresignedURLs[number],
 ) =>
-  Effect.gen(function* () {
-    const file = opts.files.find((f) => f.name === presigned.fileName);
-
-    if (!file) {
-      console.error("No file found for presigned URL", presigned);
-      return yield* new UploadThingError({
-        code: "NOT_FOUND",
-        message: "No file found for presigned URL",
-        cause: `Expected file with name ${presigned.fileName} but got '${opts.files.join(",")}'`,
-      });
-    }
-
-    opts.onUploadBegin?.({ file: file.name });
-    if ("urls" in presigned) {
-      yield* uploadMultipartWithProgress(file, presigned, opts);
-    } else {
-      yield* uploadPresignedPostWithProgress(file, presigned, opts);
-    }
-
-    const PollingResponse = S.Union(
-      S.Struct({
-        status: S.Literal("done"),
-        callbackData: S.Any as S.Schema<TServerOutput, any>,
-      }),
-      S.Struct({ status: S.Literal("still waiting") }),
-    );
-
-    let serverData: TServerOutput | null = null;
-    if (!opts.skipPolling) {
-      serverData = yield* fetchEffJson(presigned.pollingUrl, PollingResponse, {
+  Array.findFirst(opts.files, (_) => _.name === presigned.fileName).pipe(
+    Effect.tapError(() =>
+      Console.error("No file found for presigned URL", presigned),
```

Comment:
what's the difference with `Console.error` and `Effect.logError`?

Reply:
I was sure I answered you, sorry...

the `Console` module uses the users `Console` (99/100 it's just the `console`), meaning it prints the input and thats that.

The `Effect.log*` variants use the `Logger`, which is the full logging solution for effect. It has access to much more information like time, spans, annotations, and other such things logs it wherever.
Basically:
```ts
const obj = { a: 1 }

Console.log(obj) // { a: 1 }
Effect.logInfo(obj) // time="2024-05-02T08:30:00Z" level=info message="\{ \"a\": 1 \}" fiberId=1 session=abcde
```
You can add any arbitrary number of loggers, so for example a tracer (reports the logs as events on the current span) or a file-logger producing logFmt entries or some log aggregator like LogStash for chewing through arbitrary json.

TL;DR
debugging with `Console`, reporting with `Logger` (through `Effect.log*`)

Reply:
So are we doing it wrong here? https://github.com/pingdotgg/uploadthing/blob/e976fbecbe1e760c72827e6faefac7d783ee4df0/packages/uploadthing/src/internal/handler.ts#L62-L63

Reply:
Depending on what you want to achieve, if you want to invoke `console.log` and friends directly but not have control over the log level then yes, if you want more robust logging with all the bells and whistles then this is the proper way. Considering this comment:
https://github.com/pingdotgg/uploadthing/blob/e976fbecbe1e760c72827e6faefac7d783ee4df0/packages/uploadthing/src/internal/logger.ts#L141

I'd reckon that your goal would align more with the `Logger` rather than the `Console`, even if _currently_ it's being implemented in the same way as `Console` would be used. Note that `Logger` does the same thing you use `Consola` for
https://github.com/pingdotgg/uploadthing/blob/e976fbecbe1e760c72827e6faefac7d783ee4df0/packages/uploadthing/src/internal/logger.ts#L86-L106

You can replace it with `Logger` just fine. `Logger` instances have a `.log` method exposed by them so you can use them outside of effect as well. Though you would want a util for logging outside of effect since the exposed `.log` is quite demanding.
```ts
export const log = (logLevel: LogLevel.LogLevel, message: unknown) =>
  MyLogger.log({
    message,
    logLevel,
    annotations: HashMap.empty(),
    date: new Date(),
    cause: Cause.empty,
    spans: List.empty(),
    context: FiberRefs.empty(),
    fiberId: FiberId.none,
  });
```


TL;DR
Nah you're gucci, but I would shuffle some things around in that logger. I just changed a `console.error` to a `Console.error` but it should be `Effect.logError` to actually use the logger for printing